uboot-envtools: fix ipq platform ubootenv sectors size not hex format

angliu-muxi opened the pull request:
uboot-envtools: fix ipq platform ubootenv sectors size not hex format

ubootenv_size = mtd_erase * sectors, all values must in hexadecimal, but ipq platform sectors is not hex value. When fw_setenv erases the flash, the calculated erase length is greater than the actual length. If it is greater than the length of a flash block, the flash driver will report an error

Signed-off-by: Ang Liu <ang.liu@muxi-connect.com>

neheb commented on package/boot/uboot-envtools/files/ipq40xx:
```diff
@@ -25,7 +25,7 @@ ubootenv_mtdinfo () {
 	fi
 
-	sectors=$(( $ubootenv_size / $mtd_erase ))
+	sectors=$(printf 0x%x $(( $ubootenv_size / $mtd_erase )))
```
$ in $(()) is pointless. Shellcheck warns on this.

angliu-muxi replied:
thinks, fix it

neheb commented:
ping @Ansuel

Ansuel commented:
@angliu-muxi each commit require a commit description also can you provide clear example of what this fixes?

angliu-muxi commented:
> @angliu-muxi each commit require a commit description also can you provide clear example of what this fixes?

ubootenv_size = mtd_erase * sectors, all values must in hexadecimal, but ipq platform sectors is not hex value. When fw_setenv erases the flash, the calculated erase length is greater than the actual length. If it is greater than the length of a flash block, the flash driver will report an error

Ansuel commented:
@angliu-muxi ok can you put that in the commit description ?

angliu-muxi commented:
> @angliu-muxi ok can you put that in the commit description ?

ok, edited

Ansuel commented:
@angliu-muxi you need to edit the commit description not the pr description

angliu-muxi committed:
uboot-envtools: fix ipq platform ubootenv sectors size not hex format

ubootenv_size = mtd_erase * sectors, all values must in hexadecimal, but
ipq platform sectors is not hex value. When fw_setenv erases the flash,
the calculated erase length is greater than the actual length. If it is
greater than the length of a flash block, the flash driver will report
an error

Signed-off-by: Ang Liu <ang.liu@muxi-connect.com>

angliu-muxi commented:
> @angliu-muxi you need to edit the commit description not the pr description

ok, I re-edited the commit description

angliu-muxi commented:
already fixed by this commit: https://github.com/openwrt/openwrt/commit/8d3e932b65b59ab7231cb5440866eb975bd150ea
